File: app/src/main/java/com/nearsoft/nearbooks/models/realm/Book.java
package com.nearsoft.nearbooks.models.realm;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.Required;

/**
 * Book class.
 * Created by epool on 11/18/15.
 */
public class Book extends RealmObject {
    public static final String ISBN = "isbn";
    public static final String TITLE = "title";

    @PrimaryKey
    private String id;
    @Required
    private String isbn;
    @Required
    private String title;
    @Required
    private String author;
    @Required
    private String year;
    private boolean isAvailable;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getYear() {
        return year;
    }

    public void setYear(String year) {
        this.year = year;
    }

    public boolean getIsAvailable() {
        return isAvailable;
    }

    public void setIsAvailable(boolean isAvailable) {
        this.isAvailable = isAvailable;
    }
}

File: app/src/main/java/com/nearsoft/nearbooks/models/realm/Book.kt
package com.nearsoft.nearbooks.models.realm

import com.google.gson.annotations.SerializedName

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

/**
 * Realm book model.
 * Created by epool on 5/1/16.
 */
open class Book : RealmObject() {

    companion object {
        const val ID = "id"
        const val TITLE = "title"
        const val AUTHOR = "id"
        const val RELEASE_YEAR = "id"
    }

    @PrimaryKey
    @SerializedName("bookID")
    var id: String? = null

    @SerializedName("title")
    var title: String? = null

    @SerializedName("author")
    var author: String? = null

    @SerializedName("releaseYear")
    var releaseYear: String? = null

    @SerializedName("description")
    var description: String? = null

    @SerializedName("copies")
    var numberOfCopies: Int = 0

    @SerializedName("days")
    var numberOfDaysAllowedForBorrowing: Int = 0

    @SerializedName("isAvailable")
    var isAvailable: Boolean = false

    @SerializedName("borrows")
    var borrows: RealmList<Borrow>? = null

}
